test/support/utils_invoke.h

```diff
@@ -383,22 +386,21 @@ struct invoke_on_all_hetero_policies
             // performs some checks that fail. As a workaround, define for functors which have this issue
             // __functor_type(see kernel_type definition) type field which doesn't have any pointers in it's name.
             iterator_invoker<std::random_access_iterator_tag, /*IsReverse*/ std::false_type>()(
-                my_policy, op, std::forward<Args>(rest)...);
-
-
-#if TEST_CHECK_COMPILATION_WITH_COMMA_OP_DELETED_ITERS
-            // Check compilation of the kernel with comma operator deleted iterators. Use policy wrapped with unique
-            // identifier to differentiate from other calls, but, preserve the policy for other two calls.
-            TestUtils::check_compilation_no_comma(CLONE_TEST_POLICY_IDX(my_policy, 0), op, rest...);
```

Comment:
For my understanding, this could cause issues because in line 397 we pass `my_policy` directly and the test may wrap a policy similarly with `CLONE_TEST_POLICY_IDX(my_policy, 0)`?

Reply:
Yes, exactly.  Its sneaky, but we need all calls at the same level of the call stack to be wrapped with the same recursive level of cloning if we are just using numbered policy clones. If we were providing some unique name or something, that would be different.